### ISSUE TITLE
Favicon defaulting to dub.co favicon in Slack

### ISSUE THREAD

**Issue by nkkko** [dubinc/dub#709]
When sharing short links in Slack, favicon defaults to dub.co favicon like in the screenshot:
![image](https://github.com/dubinc/dub/assets/6250945/76321b92-4349-4d19-ab40-a4826f5ac6ae)

Slack doesn't respect the page's icon head tags and takes the `/favicon.ico` of the root domain.

Defaulting back to the workspace's logo could be a solution.

**Comment by versecafe** [comment]
@nkkko Wouldn't this be considered a slack issue with slack grabbing icons off the domain directly instead of following through the redirect to the end route?

**Comment by steven-tey** [comment]
@versecafe yeah that was my first impression as well! we could potentially polyfill this somehow, but would need to dig deeper into it

**Comment by versecafe** [comment]
@steven-tey I don't know if NextJS has a great way to swap the favicon programmatically BUT using a middleware layer with `NextResponse.rewrite` does work for favicon rewrites, I don't know how you'd want to handle `icon.*` vs `favicon.ico` vs custom naming conventions, you could always quick parse the <meta> tags of the end site but that takes a lot of time so I'm not sure how good of a solution it is but it does work

**Comment by shabinder-singh** [comment]
Is fixing this feasible ?
